### PR TITLE
veryfasttree: new port

### PR DESCRIPTION
**Pull request by cesarpomar**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.6 22G120 x86_64
Command Line Tools 15.0.0.0.1.1694021235

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: science/veryfasttree/Portfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+github.setup        citius veryfasttree 4.0.3 v
+github.tarball_from archive
+revision            0
+
+categories          science
+license             GPL-3
+platforms           darwin
```
Comment (reneeotten) [inline science/veryfasttree/Portfile:13]: this is the default since MacPorts v2.7 and can thus be removed

File: science/veryfasttree/Portfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+maintainers         usc.es:cesaralfredo.pineiro
+description         Efficient phylogenetic tree inference for massive taxonomic datasets
+long_description    {*}${description}
+homepage            https://github.com/citiususc/veryfasttree
```
Comment (reneeotten) [inline science/veryfasttree/Portfile:17]: this is set by the `github` PG based on the `github.setup` line and is thus redundant, please remove

File: science/veryfasttree/Portfile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+PortGroup           github 1.0
+
+github.setup        citius veryfasttree 4.0.3 v
+github.tarball_from archive
+revision            0
+
+categories          science
+license             GPL-3
+platforms           darwin
+maintainers         usc.es:cesaralfredo.pineiro
```
Comment (reneeotten) [inline science/veryfasttree/Portfile:14]: the maintainers line is usual `{obfuscated_email @GitHub-handle} openmaintainer`; the latter is to allow others to make minor changes/update to the Portfile without your permission

File: science/veryfasttree/Portfile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+homepage            https://github.com/citiususc/veryfasttree
+
+checksums           rmd160  a4070bf0ddc4aa0448f9a7878a3587a9250197a0 \
+                    sha256  770e41773806c65e60ebc77363beccf40e3d339affd97c07eb7b2e1acb51af60 \
+                    size    898112
+
+depends_lib         port:libomp
+
+configure.args-append -DUSE_NATIVE=OFF
+configure.cxxflags-append "-I${prefix}/include/libomp -L${prefix}/lib/libomp"
```
Comment (reneeotten) [inline science/veryfasttree/Portfile:26]: to get the `OpenMP` part to work you might want to check the "CompilerSelection" entry in the [wiki](https://trac.macports.org/wiki/CompilerSelection).
I *think* adding `compiler.openmp_version` with the correct value should probably be all that is needed as the dependency of `libomp` and adding of flags would happen automatically, but please check.

File: science/veryfasttree/Portfile [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    if {${build_arch} eq "i386" || ${build_arch} eq "x86_64"} {
+        configure.args-append -DUSE_AVX2=ON
+    }
+}
+
+variant AVX512 description {Add AVX extensions} {
+    if {${build_arch} eq "i386" || ${build_arch} eq "x86_64"} {
+        configure.args-append -DUSE_AVX512=ON
+    }
+}
```
Comment (reneeotten) [inline science/veryfasttree/Portfile:44]: are these options listed in the variants here detected at configure/runtime? If so, then the usual way would be to add an `native` variant and just set all of the options to ON. You should be able to `grep` for `variant native` and find other ports doing the same (often, `OpenMP` is involved in such ports as well so perhaps you can check some of the syntax for enabling that in this port as well).

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks for the submission @cesarpomar ! I haven't looked at the actual package nor tried to build anything locally, but just noted a few things for your consideration by reading through the Portfile.

File: science/veryfasttree/Portfile (hunk 1 above)
Comment (cesarpomar) [reply science/veryfasttree/Portfile]: platforms removed, done

File: science/veryfasttree/Portfile (hunk 2 above)
Comment (cesarpomar) [reply science/veryfasttree/Portfile]: homepage removed, done

File: science/veryfasttree/Portfile (hunk 3 above)
Comment (cesarpomar) [reply science/veryfasttree/Portfile]: openmaintainer and @GitHub-handle added, done

File: science/veryfasttree/Portfile (hunk 4 above)
Comment (cesarpomar) [reply science/veryfasttree/Portfile]: I used "compiler.openmp_version 2.5" but I had to add "configure.cxxflags-append -fopenmp" to make it work. The find_package(OpenMP) of cmake seems not to do it automatically in this case. I hope this workaround is acceptable.

File: science/veryfasttree/Portfile (hunk 5 above)
Comment (cesarpomar) [reply science/veryfasttree/Portfile:28]: Yes good point, I replaced all variants with a native.

**Comment by cesarpomar** [comment]
why does the bot try to download from 'Attempting to fetch veryfasttree-4.0.3.tar.gz from https://github.com/citius/veryfasttree/archive/v4.0.3' instead of 'https://github.com/citiususc/veryfasttree/archive/refs/tags/v4.0.3.tar.gz'?

File: science/veryfasttree/Portfile [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+github.setup        citius veryfasttree 4.0.3 v
```
Comment (reneeotten) [inline science/veryfasttree/Portfile:7]: there is a type on the username here, it should be `citiususc` as far as I can tell

**Comment by reneeotten** [comment]
> why does the bot try to download from 'Attempting to fetch veryfasttree-4.0.3.tar.gz from https://github.com/citius/veryfasttree/archive/v4.0.3' instead of 'https://github.com/citiususc/veryfasttree/archive/refs/tags/v4.0.3.tar.gz'?

it appears to me there is a typo in the github address: https://github.com/macports/macports-ports/pull/20837/files#r1357669474

**Comment by cesarpomar** [comment]
> > why does the bot try to download from 'Attempting to fetch veryfasttree-4.0.3.tar.gz from https://github.com/citius/veryfasttree/archive/v4.0.3' instead of 'https://github.com/citiususc/veryfasttree/archive/refs/tags/v4.0.3.tar.gz'?
> 
> it appears to me there is a typo in the github address: https://github.com/macports/macports-ports/pull/20837/files#r1357669474

Thanks

**Review by reneeotten — APPROVED** [review]

**Comment by reneeotten** [comment]
thanks @cesarpomar !

**Pull request merged**